pkg/ingress/model_build_load_balancer.go

```diff
@@ -88,31 +89,28 @@ func (t *defaultModelBuildTask) buildLoadBalancerSpec(ctx context.Context, liste
 var invalidLoadBalancerNamePattern = regexp.MustCompile("[[:^alnum:]]")
 
 func (t *defaultModelBuildTask) buildLoadBalancerName(_ context.Context, scheme elbv2model.LoadBalancerScheme) (string, error) {
-	explicitNames := sets.String{}
-	for _, member := range t.ingGroup.Members {
-		rawName := ""
-		if exists := t.annotationParser.ParseStringAnnotation(annotations.IngressSuffixLoadBalancerName, &rawName, member.Ing.Annotations); !exists {
-			continue
-		}
-		explicitNames.Insert(rawName)
-	}
-	if len(explicitNames) == 1 {
-		name, _ := explicitNames.PopAny()
-		// The name of the loadbalancer can only have up to 32 characters
-		if len(name) > 32 {
-			return "", errors.New("load balancer name cannot be longer than 32 characters")
-		}
-		return name, nil
+	explicitName, err := t.getNameAnnotation(annotations.IngressSuffixLoadBalancerName, maximumLoadBalancerNameLength)
+
+	if explicitName != "" || err != nil {
+		return explicitName, err
 	}
-	if len(explicitNames) > 1 {
-		return "", errors.Errorf("conflicting load balancer name: %v", explicitNames)
```

Comment:
Since this block of code would have to be duplicated for the prefix annotation, I just moved it to a new `getNameAnnotation` and parameterized the annotation name.